src/com/selftest/queue/AbstractQueue.java

```diff
@@ -0,0 +1,56 @@
+
+package com.selftest.queue;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public abstract class AbstractQueue<T> implements Queue<T> {
+
+	protected T[] queue;
+	protected int putLoc, getLoc;
+
+	@SafeVarargs
+	public AbstractQueue( int size, T ... values ) {
+		queue = Arrays.copyOf( values, size );
+		putLoc = getLoc = 0;
+	}
+
+	@Override
+	public synchronized Iterator<T> iterator() {
+		return new Iterator<T>() {
+
+			int position = 0;
+
+			@Override
+			public boolean hasNext() {
+				return position < queue.length;

```

Comment:
Yes, it will as my intention was to show the empty/deleted indices too (for no other reason than it being strangely satisfying to see the queue half full/empty).  

Fair point - I could/should have just used the List::iterator() method.